envoy/config/filter/http/jwt_authn/v2/config.proto

```diff
@@ -0,0 +1,216 @@
+
+  // If true, the request is allowed if the JWT verification fails.
+  // Default value is false; JWT verification failed request will be rejected.
+  google.protobuf.BoolValue allow_failed_jwt = 7;
+
+  // Two fields below define where to extract the JWT from an HTTP request.
+  //
+  // If no explicit location is specified, the following default locations are
+  // tried in order:
+  //
+  // 1. The Authorization header using the Bearer schema.
+  //    See `here <https://tools.ietf.org/html/rfc6750#section-2.1>`_
+  //    Example::
+  //
+  //       Authorization: Bearer <token>.
+  //
+  // 2. `access_token` query parameter.
+  //    See `here <https://tools.ietf.org/html/rfc6750#section-2.3>`_
+  //
+  //
+  // Only the first JWT token is verified per request if multiple tokens are
+  // specified.
+  //
+  // If both jwt_headers and jwt_params are specified, jwt_headers will be
+  // checked first.
+  //
+
+  // Specify the HTTP headers to extract JWT token.
+  //
+  // For examples, following config::
+  //
+  // .. code-block:: yaml
+  //
+  //   jwt_headers:
+  //   - name: x-goog-iap-jwt-assertion
+  //
+  // can be used to extract token from header::
+  //
+  //   x-goog-iap-jwt-assertion: <JWT>.
+  //
+  repeated JwtHeader jwt_headers = 8;
+
+  // JWT is sent in a query parameter. `jwt_params` represents the
+  // query parameter names.
+  //
+  // For example, if config is::
+  //
+  // .. code-block:: yaml
+  //
+  //   jwt_params:
+  //   - jwt_token
+  //
+  // The query parameter is::
+  //
+  //    /path?jwt_token=<JWT>
+  //
+  repeated string jwt_params = 9;
+}
+
+// This message specifies a header location to extract JWT token.
+// A HTTP header format is::
+//
+//    name: value_prefix<JWT>
+//
+message JwtHeader {
+  // The HTTP header name.
+  string name = 1;
+
+  // The value prefix. The value format is "prefix<token>"
+  // For example, for "Authorization: Bearer <token>",
+  // value_prefix="Bearer " with a space at the end.
+  string value_prefix = 2;
+}
+
+// This message defines a pattern to match a HTTP request.
+// A pattern is matched only if all "headers" are matched.
+message HttpMatcher {
+  // Specifies a set of headers should match on.
+  // For example::
+  //
+  // .. code-block:: yaml
+  //
+  //    headers:
+  //    - name: :method
+  //      value: OPTIONS
+  //    headers:
+  //    - name: :path
+  //      regex_match: /.*
+  //
+  repeated envoy.api.v2.route.HeaderMatcher headers = 1;
+}
+
+// This is the Envoy filter config for JSON Web Token authentication.
+// If a JWT is sucessfully verified, header "sec-istio-auth-userinfo" will
+// be added with base64 encoded JSON payload to be send to the backend.
+// [#not-implemented-hide:]
+message JSONWebTokenAuthentication {
+  // List of JWT rules to valide.
+  //
+  // If the request includes a JWT, it must match one of the JWT listed
+  // here with the same issuer.
+  //
+  repeated JsonWebToken jwts = 1;
```

Comment:
This is not really a JWT, it is more like JwtRule.

Reply:
Done